File: java/builder/EmailNotification.java
package builder;

public class EmailNotification {

    private final String content;
    private final String timestamp;
    private final String sender;
    private final String recipient;

    private EmailNotification(EmailNotificationBuilder emailNotificationBuilder) {
        this.content = emailNotificationBuilder.getContent();
        this.sender = emailNotificationBuilder.getSender();
        this.timestamp = emailNotificationBuilder.getTimestamp();
        this.recipient = emailNotificationBuilder.getRecipient();
    }

    public String getContent() {
        return content;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public String getSender() {
        return sender;
    }

    public String getRecipient() {
        return recipient;
    }

    public static class EmailNotificationBuilder implements NotificationBuilder {
        private String content;
        private String timestamp;
        private String sender;
        private String recipient;

        @Override
        public EmailNotificationBuilder setContent(String content) {
            this.content = content;
            return this;
        }

        @Override
        public EmailNotificationBuilder setSender(String sender) {
            this.sender = sender;
            return this;
        }

        @Override
        public EmailNotificationBuilder setRecipient(String recipient) {
            this.recipient = recipient;
            return this;
        }

        @Override
        public EmailNotificationBuilder setTimestamp(String timestamp) {
            this.timestamp = timestamp;
            return this;
        }

        public EmailNotification build() {
            return new EmailNotification(this);
        }

        public String getContent() {
            return content;
        }

        public String getTimestamp() {
            return timestamp;
        }

        public String getSender() {
            return sender;
        }

        public String getRecipient() {
            return recipient;
        }
    }
}

File: C#/Builder/EmailNotification.cs
using System;

namespace Builder
{
    public class EmailNotification
    {
        private EmailNotification(EmailNotificationBuilder emailNotificationBuilder)
        {
            Content = emailNotificationBuilder.Content;
            Sender = emailNotificationBuilder.Sender;
            Timestamp = emailNotificationBuilder.Timestamp;
            Recipient = emailNotificationBuilder.Recipient;
        }

        public string Content { get; }
        public string Timestamp { get; }
        public string Sender { get; }
        public string Recipient { get; }

        public interface NotificationBuilder
        {
            EmailNotificationBuilder SetContent(string content);
            EmailNotificationBuilder SetSender(string sender);
            EmailNotificationBuilder SetRecipient(string recipient);
            EmailNotificationBuilder SetTimestamp(string timestamp);
        }

        public class EmailNotificationBuilder : NotificationBuilder
        {
            public string Content { get; private set; }
            public string Timestamp { get; private set; }
            public string Sender { get; private set; }
            public string Recipient { get; private set; }

            public EmailNotificationBuilder SetContent(string content)
            {
                Content = content;
                return this;
            }

            public EmailNotificationBuilder SetSender(string sender)
            {
                Sender = sender;
                return this;
            }

            public EmailNotificationBuilder SetRecipient(string recipient)
            {
                Recipient = recipient;
                return this;
            }

            public EmailNotificationBuilder SetTimestamp(string timestamp)
            {
                Timestamp = timestamp;
                return this;
            }

            public EmailNotification Build()
            {
                return new EmailNotification(this);
            }
        }
    }
}
